### PR TITLE
Add test for `dv(0:*)`

### PR DESCRIPTION
**Pull request by Pranavchiku**
Refer: https://github.com/lfortran/lfortran/issues/2198#issuecomment-1692181267

### CODE REVIEW

**Comment by Pranavchiku** [comment]
The tests pass for me locally.

**Comment by Pranavchiku** [comment]
The build fails for gfortran as well, https://github.com/lfortran/lfortran/pull/2243#issuecomment-1693354732

File: integration_tests/arrays_33.f90 [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+dimension dv(0:*)
+dv(0) = 1.0D0
+end subroutine
+
+program arrays_33
+double precision :: dv(0:100)
+call pbdv(dv)
+print *, dv(0)
+if (abs(dv(0)-1.0D0) > 1.0D-10) error stop
+end program
```
Comment (Shaikh-Ubaid) [inline integration_tests/arrays_33.f90:12]: **Minor:**

```suggestion
subroutine pbdv(dv)
    implicit double precision (a-h,o-z)
    dimension dv(0:*)
    dv(0) = 1.0D0
end subroutine

program arrays_33
    double precision :: dv(0:100)
    call pbdv(dv)
    print *, dv(0)
    if (abs(dv(0)-1.0D0) > 1.0D-10) error stop
end program
```

PS: Thank you for printing the value before the assert.

Comment (certik) [reply integration_tests/arrays_33.f90:12]: Is the bug in handling of implicit typing? If not, I would not use implicit typing. The indentation is fine either way (I like it without indentation, but a lot of people like indenting it).

**Comment by Pranavchiku** [comment]
We do not need this now.

**Pull request closed without merging**